Sources/SQLiteData/CloudKit/Internal/Triggers.swift

```diff
@@ -214,19 +217,48 @@
       createTemporaryTrigger(
         "after_update_on_sqlitedata_icloud_metadata",
         ifNotExists: true,
-        after: .update { _, new in
+        after: .update { old, new in
+          let zoneChanged = new.zoneName.neq(old.zoneName) || new.ownerName.neq(old.ownerName)
+          let selfAndDescendantRecordNames = descendantRecordNames(
+            recordName: new.recordName,
+            includeRecord: true
+          ) {
+            $0.select(\.recordName)
+          }
+          let descendantRecordNamesJSON = descendantRecordNames(
+            recordName: new.recordName,
+            includeRecord: false
+          ) {
+            $0.select { $0.recordName.jsonGroupArray() }
+          }
+
           validate(recordName: new.recordName)
+          SyncMetadata
+            .where {
+              zoneChanged && $0.recordName.in(selfAndDescendantRecordNames)
+            }
+            .update {
+              $0.zoneName = new.zoneName
+              $0.ownerName = new.ownerName
+              $0.lastKnownServerRecord = nil
+              $0._lastKnownServerRecordAllFields = nil
+            }
           Values(
             syncEngine.$didUpdate(
               recordName: new.recordName,
-              lastKnownServerRecord: new.lastKnownServerRecord
-                ?? rootServerRecord(recordName: new.recordName),
-              newParentLastKnownServerRecord: parentLastKnownServerRecordIfShared(
-                parentRecordPrimaryKey: new.parentRecordPrimaryKey,
-                parentRecordType: new.parentRecordType
-              ),
-              parentRecordPrimaryKey: new.parentRecordPrimaryKey,
-              parentRecordType: new.parentRecordType
+              zoneName: new.zoneName,
+              ownerName: new.ownerName,
+              oldZoneName: old.zoneName,
+              oldOwnerName: old.ownerName,
+              childRecordNames: #sql(
+                """
+                iif(
+                  \(zoneChanged),
+                  \(descendantRecordNamesJSON),
+                  NULL
+                )
+                """
+              )
```

Comment:
Could this be the following?
```suggestion
              childRecordNames: Case().when(zoneChanged, then: descendantRecordNamesJSON)
                """
              )
```